src/mqt/qecc/circuit_synthesis/operations.py

```
@@ -0,0 +1,771 @@
+
+    def __init__(self, qubit: int, pauli: str) -> None:
+        """Initialize the Pauli operation.
+
+        Args:
+            qubit: The index of the qubit.
+            pauli: The Pauli operation to apply {X, Y, Z}.
+        """
+        self.qubit = qubit
+        self.pauli = pauli
+
+    def apply_check_matrix(self, check_matrix: CheckMatrix, inplace: bool = False) -> CheckMatrix:  # noqa: ARG002, PLR6301
```

Ruff (0.15.12)

[warning] Boolean-typed positional argument in function definition

(FBT001)